bin/tx-prover/src/proxy/mod.rs

```diff
@@ -0,0 +1,193 @@
+use std::{collections::HashMap, sync::Arc, time::Duration};
+
+use async_trait::async_trait;
+use once_cell::sync::Lazy;
+use pingora::{
+    http::ResponseHeader,
+    lb::Backend,
+    prelude::*,
+    upstreams::peer::{Peer, ALPN},
+};
+use pingora_core::{upstreams::peer::HttpPeer, Result};
+use pingora_limits::rate::Rate;
+use pingora_proxy::{ProxyHttp, Session};
+use tokio::sync::RwLock;
+use tracing::error;
+
+/// Timeout for the requests
+const TIMEOUT_SECS: Option<Duration> = Some(Duration::from_secs(100));
+/// Maximum number of items per queue
+const MAX_QUEUE_ITEMS: usize = 10;
+
+/// Load balancer that uses a round robin strategy
+pub struct LB(Arc<LoadBalancer<RoundRobin>>);
```

Comment:
nit: I would change the name of this struct. Maybe something like `WorkerLoadBalancer`? Or `RoundRobinBalancer` if we want it to be more general?